apps/website/docs/magazine/migration_from_redux.md

```diff
@@ -0,0 +1,308 @@
+  filter: (name) => name.length <= 20,
+  target: $userName,
+});
+```
+
+☝️ Feature is completely ported to Effector, `reduxInterop` is not used here anymore.
+
+If there is still code that consumes this state via the Redux Store selector, and there is currently no way to move that consumer to use the Effector model, it is still possible to "sync" the state back into Redux as a read-only mirror of the Effector model state:
+
+```ts
+// src/features/user-info/model.ts
+
+// ...main code
+
+// sync state back to Redux
+sample({
+  clock: $userName,
+  target: [
+    reduxInterop.dispatch.prepend((name: string) =>
+      userInfoSlice.syncNameFromEffector(name)
+    ),
+  ],
+});
+```
+
+☝️ But it's important to make sure that this is a read-only mirror that won't be changed in Redux in any other way - because then there would be two parallel versions of this state, which would probably lead to nasty bugs.
+
+## New feature
+
+Adding a new feature on Effector to a Redux project is not much different from the initial step of migrating an existing feature:
+
+1. Any new code is written in Effector
+2. Any dependencies to Redux Store should work through `reduxInterop` API
+
+## Special cases
+
+### Middleware with side-effects
+
+Sometimes Redux actions are not changing state, but trigger side-effects via middlewares.
+
+Suppose Redux Store has middleware that reacts to action like `{ type: SEND_ANALYTICS_EVENT, payload }` and sends the event to our analytics.
+
+Sending analytics is usually involved in almost all code of the application and migration of such a feature will be much more complicated.
+
+In this case, the recommended upgrade path is as follows:
+
+#### Mirror of the action
+
+First, create a mirror Effector's event of the `SEND_ANALYTICS_EVENT` action by using its action-creator:
+
+```ts
+// src/shared/analytics/model.ts
+import { reduxInterop } from 'root/redux-store';
+import { sendAnalyticsEventAction } from './actions';
+
+export const sendAnalytics = reduxInterop.dispatch.prepend((payload) =>
+  sendAnalyticsEventAction(payload)
+);
+```
+
+#### Move to event instead of an action
+
+As a second step, gradually change all dispatches of this action to a event call.
+
+E.g. instead of
+
+```ts
+import { sendAnalyticsEventAction } from 'root/analytics';
+
+dispatch(sendAnalyticsEventAction(payload));
+```
+
+do
+
+```ts
+import { sendAnalytics } from 'root/analytics';
+
+sendAnalytics(payload);
+```
+
+It is safe to do, because the `sendAnalytics(payload)` call here is a full equivalent of the `dispatch(sendAnalyticsEventAction(payload))` and can be used instead of it - the action will still be dispatched by the `reduxInterop.dispatch` under the hood.
+
+In the end Redux, Effector and your UI-framework should all use this event instead of dispatching the action.
+
+#### Move the implementation
+
+Since now all analytics is sent via this event, it is now possible to fully move from a analytics middleware to Effector's model:
```

Comment:
```suggestion
Since now all analytics is sent via this event, it is now possible to fully move from the analytics middleware to Effector's model:
```